### ISSUE TITLE
Cannot evaluate a 'lat_lon' DataSource with 'lat' requested coordinates

### ISSUE THREAD

**Issue by jmilloy** [creare-com/podpac#123]
## Description
When a DataSource has stacked native coordinates, it should be possible in some cases to evaluate it even if some of the dimensions are missing.

### native_coordinates `lat_lon_time`, requested coordinates `lat_lon` or `time`

```
node = Array(
    source=np.ones(3),
    native_coordinates=Coordinates([[[0, 2, 1], [10, 12, 11], ['2018-01-01', '2018-01-02', '2018-01-03']]], dims=['lat_lon_time']))

# unstacked or and stacked requests without time
o1 = node.eval(Coordinates([[0.5, 1.5], [10.5, 11.5]], dims=['lat', 'lon']))
o2 = node.eval(Coordinates([[[0.5, 1.5], [10.5, 11.5]]], dims=['lat_lon']))

# request without lat or lon
o3 = node.eval(Coordinates(['2018-01-01'], dims=['time']))
```

## Notes
Currently, the DataSource eval checks the requested coordinates against the native coordinates and will raise an exception in cases where there is a missing stacked dimension. The commit c75babc reverted a version of these checks that allows missing stacked dimensions if they are monotonic. In fact, the missing stacked dimension may not need to be monotonic, you could sort by that dimension during interpolation. Both the DataSource eval and the interpolation will need to be updated to correctly handle these cases correctly.

**Comment by jmilloy** [comment]
@mlshapiro We may choose to handle this in the interpolation, but I might be able to fix it in the eval without you having to touch the interpolation, so for the time being I've assigned both of us.

**Comment by jmilloy** [comment]
@mlshapiro No, after a better look, I think this is a case that needs to be handled during the interpolation. Let me know if this isn't making sense or if you think I've overlooked something.

**Comment by jmilloy** [comment]
(currently only in the feature/node_spec branch, I'll update once this is merged in)

**Comment by mlshapiro** [comment]
i will take a look at this today when I get through the full implementation of nearest and nearest_preview. thanks for tracking.

**Comment by jmilloy** [comment]
Ah, I removed the need for this in c75babc.

**Comment by mpu-creare** [comment]
Why did you remove the need for this? It's a feature that I need and really want implemented in podpac...

**Comment by mpu-creare** [comment]
That's fair. My use case has:
* lat-lon-time triplet (rover data collecting measurements). 
* time only dataset (another sensor that isn't georegistered)
* static lat,lon gridded datasets

So I want to be able to use the coordinates of the the second and third bullet to index into the first bullet. 

In fact, I also want to be able to use the coordinates of the first bullet to index into the second and third bullet.

**Comment by jmilloy** [comment]
Okay, great. The test removed in c75babc didn't cover this case, nor did this issue as I initially write it. I've updated the description.

**Comment by mpu-creare** [comment]
This will be fixed once #430 is merged.
